engine/src/main/java/io/camunda/zeebe/engine/processing/bpmn/behavior/BpmnDecisionBehavior.java

```diff
@@ -76,7 +80,12 @@ public BpmnDecisionBehavior(
       final ExecutableCalledDecision element, final BpmnElementContext context) {
     final var scopeKey = context.getElementInstanceKey();
 
-    final var decisionId = element.getDecisionId().getExpression();
+    final var decisionIdOrFailure = evalDecisionIdExpression(element, scopeKey);
+    if (decisionIdOrFailure.isLeft()) {
+      return decisionIdOrFailure.map(null);
```

Comment:
❓ Why does this return a `decisionIfOrFailure.map(null)`? It seems like this is just a trick to fix types of the `Either`, correct?

Reply:
Yes that is correct. The type of decisionIdOrFailure is `Either<Failure, String>` but the return type expects an `Either<Failure, DecisionEvaluationResult>`.

The only alternative I know is to wrap the left's value into a new left, but that's less functional. I this case it would be `return Either.left(decisionIdOrFailure.getLeft());`. 

After having worked with Eithers for some time, I'm still not sure which I like better. My preference goes slightly to the map. The map looks strange at first but is really concise and accurate. We simply map the right side of a left to the correct type and `null` makes it clear that we don't know its value. But feel free to counter this. I'm also open to other ideas.

Reply:
As someone who has barely worked with `Either` the map seems very weird. You're mapping something, but because it's a left you won't actually map anything. I would expect this code to throw a NPE at first since it seems like it applies the given function to something, but in reality the entire `map()` method doesn't do anything because it's a left. So I think my preference goes to `Either.left(decisionIdOrFailure.getLeft())`. Then again it's not a very strong preference since it's all documented quite well.

A separate method which would do this for us to make it more readable would be nice. But the problem is that it would then be on `Either`, and `Right` would need an implementation of it which is not really possible. It's a hard problem 😄

Reply:
Also I do like the more functional approach

Reply:
TBH, you're absolutely right. And it should be clear for first-time readers. I've changed it to the non-functional style. 

I've also looked around our codebase for similar situations and found 2 cases that used Either.left(x.getLeft()) and 1 case that uses map. 😆 Nothing really helpful. This other map case can't be changed though, since it's really mapping a right as well. See https://github.com/camunda-cloud/zeebe/blob/main/engine/src/main/java/io/camunda/zeebe/engine/processing/common/CatchEventBehavior.java#L104